### PR TITLE
Replaced editor, config file location with cross-compile options

### PR DESCRIPTION
**Pull request by vedangj044**
Work in progress for #19 

- The XDG path specified to store the config file is calculated on the basis of the underlying OS using this crate - [dirs-rs](https://crates.io/crates/dirs)
- The editor required to use the "edit"  command is opened using this crate - [edit](https://crates.io/crates/edit)

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -194,13 +194,14 @@ impl Context {
...
-// todo crossplatform support
 fn eval_config_path(path: Option<&str>) -> String {
     match path {
         Some(p) => p.to_string(),
         None => env::var("DRGCFG").unwrap_or_else(|_| {
-            let xdg = env::var("XDG_CONFIG_HOME")
-                .unwrap_or(format!("{}/.config", env::var("HOME").unwrap()));
+            let xdg = match config_dir() {
+                Some(path) => path.into_os_string().into_string().unwrap(),
+                None => String::from("."),
```
Comment (vedangj044) [inline src/config.rs:203]: Please specify a default path

Comment (jbtrystram) [reply src/config.rs:203]: I would error out if a path cannot be found. The error message should suggest the user to use the `--config` option

**Review by jbtrystram — CHANGES_REQUESTED** [review]

File: src/config.rs (hunk 1 above)
Comment (vedangj044) [reply src/config.rs]: Added !

File: src/util.rs [hunk 2]
```diff
@@ -86,7 +86,7 @@ pub fn editor(original: String) -> Result<Value> {
     edit::edit_file(file.path())
         .map_err(|err| {
             log::debug!("{}", err);
-            log::error!("Error opening a text editor, pelase try --file");
+            log::error!("Error opening a text editor, please try using --filename");
```
Comment (jbtrystram) [inline src/util.rs:89]: I would also print the file content to the console so the user can copy and paste it into the file

**Review by jbtrystram — CHANGES_REQUESTED** [review]

File: src/util.rs (hunk 2 above)
Comment (vedangj044) [reply src/util.rs]: Good Idea

**Pull request merged**